mne_sandbox/preprocessing/_sns.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+"""Sensor noise suppression"""
+
+import numpy as np
+
+from mne import compute_raw_covariance
+from mne.io.pick import _pick_data_channels, pick_channels
+from mne.io import _BaseRaw
+from mne.utils import logger, verbose
+
+from ._dss import _pca
+
+
+class SensorNoiseSuppression(object):
+    """Apply the sensor noise suppression (SNS) algorithm
+
+    This algorithm (from [1]_) will replace the data from each channel by
+    its regression on the subspace formed by the other channels.
+
+    .. note:: Bad channels are not modified or reset by this function.
+
+    Parameters
+    ----------
+    n_neighbors : int
```

Comment:
They call them `neighbors` in the paper. And I'm not sure why you'd call it `skip` anyway, it's the number of channels to use in the calculation...?